Doc/library/inspect.rst

```diff
@@ -694,7 +694,7 @@ and its return annotation. To retrieve a :class:`!Signature` object,
 use the :func:`!signature`
 function.
 
-.. function:: signature(callable, *, follow_wrapped=True, globals=None, locals=None, eval_str=False)
+.. function:: signature(callable, *, follow_wrapped=True, globals=None, locals=None, eval_str=False, annotation_format=Format.VALUE)
```

Comment:
This has the ironic effect of making the signature returned by `inspect.signature(inspect.signature)` not very pretty:

```pycon
>>> import inspect
>>> inspect.signature(inspect.signature)
<Signature (obj, *, follow_wrapped=True, globals=None, locals=None, eval_str=False, annotation_format=<Format.VALUE: 1>)>
```

The value of `Format.VALUE` is sort-of irrelevant; I think we should change the repr of the `Format` enum so that it's prettier in situations like this. Something like this?

```diff
--- a/Lib/annotationlib.py
+++ b/Lib/annotationlib.py
@@ -25,6 +25,9 @@ class Format(enum.IntEnum):
     FORWARDREF = 2
     STRING = 3
 
+    def __repr__(self):
+        return f"Format.{self.name}"
+
```

Then we get this:

```pycon
>>> import inspect
>>> inspect.signature(inspect.signature)
<Signature (obj, *, follow_wrapped=True, globals=None, locals=None, eval_str=False, annotation_format=Format.VALUE)>
```

Reply:
I'd prefer to use the defaults of the enum module unless there is some reason this particular enum is special. If you feel the way enums are formatted by default is not ideal, we should fix that in the enum module, not here.

Reply:
fair!